text/2539-cfg_attr-multiple-attrs.md

```diff
@@ -0,0 +1,180 @@
+author wanted those attributes configured behind the same predicate.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+The `cfg_attr` attribute takes a configuration predicate and then a list of
+attributes that will be in effect when the predicate is true.
+
+For an example of multiple attributes, say we want to have two attribute macros
+(`sparkles` and `crackles`), but only when `feature = "magic"` is enabled. We
+can write this as:
+
+```rust,igore
+#[cfg_attr(feature = "magic", sparkles, crackles)]
+fn bewitched() {}
+```
+
+When the feature flag is enabled, it expands to:
+
+```rust,ignore
+#[sparkles]
+#[crackles]
+fn bewitche() {}
+```
+
+The list of attributes may be empty, but will warn if the actual source code
+contains an empty list.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+The next section replaces what's in the Conditional Compilation Chapter for the
+`cfg_attr` attribute. It explains both current and new behavior, mainly because
+the current reference material needs improvement.
+
+## `cfg_attr` Attribute
+
+The `cfg_attr` attribute conditionally includes attributes based on a
+configuration predicate. 
+
+It is written as `cfg_attr` followed by `(`, a comma separated metaitem
+sequence, and then `)` The metaitem sequence contains one or more metaitems.
+The first is a conditional predicate. The rest are metaitems that are also
+attributes. Trailing commas after attributes are permitted. The following list
+are all allowed:
+
+* `cfg_attr(predicate, attr)`
+* `cfg_attr(predicate, attr_1, attr_2)`
+* `cfg_attr(predicate, attr,)`
+* `cfg_attr(predicate, attr_1, attr_2,)`
+* `cfg_attr(predicate,)`
+
+> Note: `cfg_attr(predicate)` is not allowed. That comma is semantically
+> distinct from the commas following attributes, so we require it.
+
+When the configuration predicate is true, this attribute expands out to be an
+attribute for each attribute metaitem. For example, the following module will
+either be found at `linux.rs` or `windows.rs` based on the target.
+
+```rust,ignore
+#[cfg_attr(linux, path = "linux.rs")]
+#[cfg_attr(windows, path = "windows.rs")]
+mod os;
+```
+
+For an example of multiple attributes, say we want to have two attribute macros,
+but only when `feature = "magic"` is enabled. We can write this as:
+
+```rust,igore
+#[cfg_attr(feature = "magic", sparkles, crackles)]
+fn bewitched() {}
+```
+
+When the feature flag is enabled, the attribute expands to:
+
+```rust,ignore
+#[sparkles]
+#[crackles]
+fn bewitche() {}
+```
+
+Note: The `cfg_attr` can expand to another `cfg_attr`. For example,
+`#[cfg_attr(linux, cfg_attr(feature = "multithreaded", some_other_attribute))`
+is valid. This example would be equivalent to
+`#[cfg_attr(and(linux, feaure ="multithreaded"), some_other_attribute)]`.
+
+## Warning When Zero Attributes
+
+This RFC allows `#[cfg_attr(predicate)]`. This is so that macros can generate
```

Comment:
This should have been `#[cfg_attr(predicate, )]`?

Reply:
Yeah, the RFC states that `#[cfg_attr(predicate)]` is not permitted before this. A fix PR would be welcome.